### PR TITLE
Added light/dark mode [Issue #97]

### PR DESCRIPTION
**Pull request by cosay**
Added light/dark mode.
Added some CSS for dark theme and JS functions for switching modes.

Issue: #97

### CODE REVIEW

File: assets/css/index.css [hunk 1]
```diff
@@ -9,6 +9,30 @@ body {
   color: #333;
 }
 
+[data-theme="dark"] {
+  background-color: #131313;
```
Comment (shahednasser) [inline assets/css/index.css:13]: add a variable in `assets/css/variables.css` to value `#262626` and use it here for background color

File: assets/css/index.css [hunk 2]
```diff
@@ -9,6 +9,30 @@ body {
...
+  background-color: #000000;
+}
+
+[data-theme="dark"] .navbar.scrolling {
+  background-color: #000000;
+  box-shadow: 0px 2px 3px #1e1e1e;
+}
+
+[data-theme="dark"] .sidebar {
+  background-color: #131313;
```
Comment (shahednasser) [inline assets/css/index.css:27]: same variable as before

**Review by shahednasser — CHANGES_REQUESTED** [review]
Please also add to body `transition: all .2s;`

on line 152, add `.toggle-theme` to the css rule as the following:
```
.navbar .download-link,
.navbar .icon-link,
.navbar .sidebar-toggler,
.navbar .toggle-theme {
  opacity: 0.5;
  transition: all 0.2s;
}
```

and then on line 157 add `.toggle-theme` as well to the rule:
```
.navbar .download-link:hover,
.navbar .icon-link:hover,
.navbar .sidebar-toggler:hover,
.navbar .toggle-theme:hover {
  opacity: 1;
}
```

File: assets/css/index.css [hunk 3]
```diff
@@ -9,6 +9,30 @@ body {
   color: #333;
 }
 
+[data-theme="dark"] {
+  background-color: #131313;
+  color: #ffffff;
+}
+
+[data-theme="dark"] footer, .navbar.scrolling {
+  background-color: #000000;
```
Comment (shahednasser) [inline assets/css/index.css:18]: Change footer background to `#434343`

Comment (cosay) [reply assets/css/index.css:18]: And the navbar stays #000000 or should I change it too?

Comment (shahednasser) [reply assets/css/index.css:18]: Change it too

**Comment by cosay** [comment]
Now should be ok :)

**Comment by shahednasser** [comment]
It looks good but I still need you to add `#262626` as a variable in `assets/css/variables.css` and replace it with the variable name.

File: assets/css/index.css [hunk 4]
```diff
@@ -163,6 +189,12 @@ hr {
 
 .navbar .download-link {
   margin-right: 1em;
+  text-decoration: none;
+}
+
+.navbar .toggle-theme {
+  margin-right: 2em;
```
Comment (shahednasser) [inline assets/css/index.css:196]: Also change this to 1em

**Comment by cosay** [comment]
> 
> 
> It looks good but I still need you to add `#262626` as a variable in `assets/css/variables.css` and replace it with the variable name.

My bad, fixed.

**Comment by shahednasser** [comment]
Great! Thank you

@all-contributors please add @cosay for design

**Pull request merged**